reader/components/pdf-outline-sidebar/renderers/OutlineRenderer.jsx

```diff
@@ -3,22 +3,24 @@ import { Collapse } from 'reactstrap'
 
 const OutlineGroup = ({ outline, linkService, isExpanded }) => {
   return (
-    <Collapse isOpen={isExpanded}>
-      <ol>
-        {outline.map((item, index) => {
-          /* eslint-disable react/no-array-index-key */
-          return (
-            <OutlineItem
-              key={`${index}${isExpanded}`}
-              item={item}
-              linkService={linkService}
-              isExpanded={isExpanded}
-            />
-          )
-          /* eslint-enable */
-        })}
-      </ol>
-    </Collapse>
+    <>
+      <Collapse isOpen={isExpanded}>
+        <ol>
+          {outline.map((item, index) => {
+            /* eslint-disable react/no-array-index-key */
+            return (
+              <OutlineItem
+                key={`${index}${isExpanded}`}
```

Comment:
If I understand React's reconciliation, it's better to have bare index only.

Reply:
Can you point me to some articles?

Reply:
Explanation of [Reconciliation in the React documentation](https://reactjs.org/docs/reconciliation.html) might be the best I can send you.

The idea of key is to help React to understand that equal elements, i.e. types are equal, are different and vice versa. In other words, React will create new node for the elements with new keys and remove nodes for elements that keys has disappeared.

Coming back to your case, when the item becomes expanded/collapsed, React removes the item HTML element from the tree, creates and places a new one. It might be somehow optimised, e.g. elements may be reused but I cannot be sure because I have not explored the code.

Reply:
Make sense. I think the reason for using ```key{`${index}${isExpanded}`}``` was to get rid of `eslint-disable`. But obviously it didn't work. 😄